homestar-runtime/src/cli.rs

```diff
@@ -34,12 +37,84 @@ Usage: {usage}
 #[derive(Debug, Parser)]
 #[command(bin_name = "homestar", name = "homestar", author, version, about,
           long_about = None, help_template = HELP_TEMPLATE)]
+#[clap(group(ArgGroup::new("init_sink").args(&["config", "dry-run"])))]
+#[clap(group(ArgGroup::new("init_key_arg").args(&["key-file", "key-seed"])))]
 pub struct Cli {
     /// Homestar [Command].
     #[clap(subcommand)]
     pub command: Command,
 }
 
+/// Arguments for `init` command.
+#[derive(Debug, Clone, PartialEq, Args)]
+pub struct InitArgs {
+    /// Runtime configuration file (.toml).
+    #[arg(
+            short = 'o',
+            long = "output",
+            value_hint = clap::ValueHint::FilePath,
+            value_name = "OUTPUT",
+            help = "Path to write initialized configuration file (.toml) [optional]",
+            group = "init_sink"
+        )]
+    pub output_path: Option<PathBuf>,
+    /// Skip writing to disk.
+    #[arg(
+        long = "dry-run",
+        help = "Skip writing to disk",
+        default_value = "false",
+        help = "Skip writing to disk, instead writing configuration to stdout [optional]",
+        group = "init_sink"
+    )]
+    pub dry_run: bool,
+    /// Suppress auxiliary output.
+    #[arg(
+        short = 'q',
+        long = "quiet",
+        default_value = "false",
+        help = "Suppress auxiliary output [optional]"
+    )]
+    pub quiet: bool,
+    /// Force destructive operations without prompting.
+    #[arg(
+        short = 'f',
+        long = "force",
+        default_value = "false",
+        help = "Force destructive operations without prompting [optional]"
+    )]
+    pub force: bool,
+    /// Run in non-interactive mode by disabling all prompts.
+    #[arg(
+        long = "no-input",
+        default_value = "false",
+        help = "Run in non-interactive mode [optional]"
+    )]
+    pub no_input: bool,
+    /// The type of key to use for libp2p
+    #[arg(
+        long = "key-type",
+        value_name = "KEY_TYPE",
+        help = "The type of key to use for libp2p [optional]"
+    )]
+    pub key_type: Option<KeyType>,
+    /// The file to load the key from
+    #[arg(
+        long = "key-file",
+        value_name = "KEY_FILE",
+        help = "The path to the key file. A key will be generated if the file does not exist [optional]",
```

Comment:
minor: add a note that a key will be generated to <default path> if no file is provided.